.github/workflows/generate-pricing-list.yml

```diff
@@ -0,0 +1,115 @@
+name: Pricing List
+
+on:
+  workflow_dispatch:
+    inputs:
+      spreadsheet_uri:
+        description: Link of the spreadsheet containing subscription details.
+        type: string
+        required: true
+      spreadsheet_key:
+        description: Key/permission to access the spreadsheet
+        type: string
+        required: true
+  schedule:
+    - cron: "0 0 * * *"
+
+jobs:
+  fetch-pricing-list:
+    name: Fetch Pricing List
+    if: github.repository == 'layer5io/docs'
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v4
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          fetch-depth: 1
+
+      - name: Setup go
+        uses: actions/setup-go@v4
+        with:
+          go-version: ${{ secrets.GO_VERSION }}
+
+      - name: Install dependencies
+        run: |
+          go get -u github.com/gocarina/gocsv
+          go get -u github.com/tidwall/gjson
+
+      - name: Fetch and process pricing list
+        env:
+          SPREADSHEET_URI: https://docs.google.com/spreadsheets/d/1Ck_5q7U_vLSIDTtplugG3pCVC5zugXgTHtO7T7-yL8g/export?format=csv&gid=0
+          SPREADSHEET_KEY: ${{ secrets.SPREADSHEET_KEY }}
+        run: |
+          go run <<EOF
```

Comment:
It would be nice to use the GitHub docker container action for this. Wdyt?

https://docs.github.com/en/actions/sharing-automations/creating-actions/creating-a-docker-container-action#creating-a-dockerfile

Example from Python code:
https://github.com/meshery/meshery/blob/master/.github/actions/comment-preview-on-pr/action.yml

Reply:
Yes, that makes a lot more sense! I'll implement that.